src/lib/y2partitioner/actions/controllers/fstabs.rb

```diff
@@ -402,7 +403,14 @@ def original_filesystem(filesystem)
         #   not found in the devicegraph.
         def entry_device(entry, devicegraph)
           device = entry.device(devicegraph_with_crypttab_names(devicegraph))
-          device ? devicegraph.find_device(device.sid) : nil
+
+          return nil unless device
+
+          if missing_swap_encryption?(device, devicegraph)
+            create_swap_encryption(device, devicegraph)
```

Comment:
@joseivanlopez, While trying to understand this code, I realize that what it does can be described as copying `device` into `devicegraph`. Since `device` comes from another (so-called fake) graph.

Maybe it would be more obvious if that would be reflected in the name of `create_swap_encryption`. Or maybe even going one step further and using `copy_to_devicegraph` so the method does exactly that (instead of creating an encryption object in `devicegraph` just to "imitate" how another equivalent encryption object was created in the fake devicegraph [here](https://github.com/yast/yast-storage-ng/pull/960/files#diff-0e53ffcdfa995a7567219ea93442be9bR285)).